json/pickler/src/main/scala/sttp/tapir/json/pickler/Pickler.scala

```diff
@@ -30,10 +30,15 @@ object Pickler:
     *
     * The in-scope [[PicklerConfiguration]] instance is used to customise field names and other behavior.
     */
-  inline def derived[T: ClassTag](using PicklerConfiguration, Mirror.Of[T]): Pickler[T] =
+  inline def derived[T: ClassTag](using PicklerConfiguration): Pickler[T] =
     summonFrom {
       case schema: Schema[T] => fromExistingSchemaAndRw[T](schema)
-      case _                 => buildNewPickler[T]()
+      case _ =>
+        summonFrom {
```

Comment:
can't we flatten the `summonFrom`s?

Reply:
Hmm looks like we can indeed. The code feels strange, but works, thanks :)